### DISCUSSION TITLE
Failing composite action verbatim from docs

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#24985] [Actions]
<p>I just tried to create a composite action, using verbatim the code from the <a href="https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action" rel="noopener nofollow ugc">documentation</a>, but I get two different error messages under windows and ubuntu runners:</p>
<p>Windows runner:</p>
<pre><code class="lang-auto">D:\a\_temp\d1b675e1-0df0-4b9d-8386-8fd25d8ff5e7.sh: line 1: D:a_actionsaleks-ivanovtest-actionmain/goodbye.sh: No such file or directory
</code></pre>
<p>Ubuntu runner:</p>
<pre><code class="lang-auto">/home/runner/work/_temp/9638bc2a-7e36-49bb-afc9-4509b4f153d0.sh: line 1: /home/runner/work/_actions/aleks-ivanov/test-action/main/goodbye.sh: Permission denied
</code></pre>
<p>I know the action is accessed and it executes, because it prints “Hello Mona the Octocat.” on both, as per this part of the action:</p>
<pre><code class="lang-auto">    - run: echo Hello ${{ inputs.who-to-greet }}.
      shell: bash
</code></pre>
<p>This is the code that is using the action:</p>
<pre><code class="lang-auto">      - id: foo
        uses: aleks-ivanov/test-action@main
        with:
          who-to-greet: 'Mona the Octocat'
</code></pre>
<p>What am I missing ?</p>

**Comment by aleks-ivanov** [comment, marked as answer]
<p>UPDATE:</p>
<p>made it work on “ubuntu-latest” by adding the “chmod +x goodbye.sh” to the action itself like this (ALTHOUGH, in the documentation it is not described that way):</p>
<pre><code class="lang-auto">    - run: |
        cd ${{ github.action_path }}
        chmod +x goodbye.sh
      shell: bash
</code></pre>
<p>However on “windows-latest”, the “cd ${{ github.action_path }}” returns and error that there is no such a directory:</p>
<pre><code class="lang-auto">D:\a\_temp\51f7870d-4631-483a-bc67-da3993b468e4.sh: line 2: cd: D:a_actionsaleks-ivanovtest-actionmain: No such file or directory
</code></pre>
<p>UPDATE 2:</p>
<p>made it work on “windows-latest” as well.</p>
<p>Turns out that “windows-latest” doesn’t work with <strong>${{ github.action_path }}</strong>, so instead I used $GITHUB_ACTION_PATH, which apparently fixes the issue:</p>
<pre><code class="lang-auto">- name: Run tool's script
      run: $GITHUB_ACTION_PATH/csabd.sh
</code></pre>

**Comment by ClydeDz** [comment]
I tried adding chmod to the same line but got this error `chmod: cannot access '<value>': No such file or directory`. 

After splitting into two steps, it worked. 

```
- run: chmod +x goodbye.sh
    shell: bash

- run: ./goodbye.sh ${{ inputs.parameter }}
    shell: bash
```
I believe the documentation should be updated to say that chmod is needed even in the action itself before running the shell script.
